### DISCUSSION TITLE
I can't get the pop up to show up

### DISCUSSION THREAD

**Discussion by joao-portelinha** [abacritt/angularx-social-login#535] [Q&A]
I've followed several tutorials, and everytime i click my button which i assigned the login funcion it just throws this error:
![image](https://user-images.githubusercontent.com/95179161/177366528-63c6c4db-8403-4f4f-8804-61f02db9606e.png)
What can i do to fix this?

**Comment by anton-nikitsiuk** [comment]
Just read this https://developers.google.cn/identity/gsi/web/guides/migration

And this library follows the guide, so no direct calls to the API, only via the GSI interface.
Just use the button `<asl-google-signin-button></asl-google-signin-button>` or create your own using new GSI API.

**Reply by adnen178** [reply]
I have the same issue, I read the docs but I can't find anything useful to integrate it into angular

**Reply by joao-portelinha** [reply]
Yeah i did not find a way to make it work... Had to used something else i found on youtube since it was for a uni assignment.
I used the package "angular-oauth2-oidc", from a video i found on youtube.

**Reply by anton-nikitsiuk** [reply]
Here is what I have, it works perfectly for me

<div #google id="google-login" class="google-login"></div>

```
import { SocialAuthService } from '@abacritt/angularx-social-login';
import { take } from 'rxjs';
declare const google: any;

@Component({
 *******
})
export class ***Component implements AfterViewInit {
  @ViewChild('google') googleDiv: ElementRef<HTMLElement>;

  constructor(
    public socialAuthService: SocialAuthService,
  ) {
  }

  ngAfterViewInit(): void {
    this.socialAuthService.initState.pipe(take(1)).subscribe(() => {
      google.accounts.id.renderButton(this.googleDiv.nativeElement, {
        type: 'standard',
        size: 'large',
        width: this.googleDiv.nativeElement.clientWidth || '100%',
        logo_alignment: 'center',
        text: 'continue_with',
        locale: 'en-US',
      });
    });
  }
```
